slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c

```diff
@@ -938,8 +936,9 @@ void setCounterMask(uint32_t arg) {
     for (int i = 0; i < NCOUNTERS; ++i) {
         // if change in enable
         if ((arg & (1 << i)) ^ (oldmask & (1 << i))) {
-            // will disable if counter disabled
-            setDAC(VTH1, vthEnabledVals[i], 0);
+            // will disable if counter disabled, else set corresponding vth dac
+            enum DACINDEX vth = (i == 0 ? M_VTH1 : (i == 1 ? M_VTH2 : M_VTH3));
```

Comment:
The double ternary operator took me a few seconds to understand =) 

maybe make something like 
DACINDEX ind[3]{M_VTH1, M_VTH2, M_VTH3}
for ...
   ind[i]

Reply:
makes sense!